### PR TITLE
Handle nested function applications in named RA translation

### PR DESCRIPTION
**Pull request by tgy**

### CODE REVIEW

File: neurolang/datalog/translate_to_named_ra.py [hunk 1]
```diff
@@ -174,25 +174,57 @@ def translate_builtin_fa_constants(self, expression):
     @add_match(FunctionApplication)
     def translate_fa(self, expression):
         functor = self.walk(expression.functor)
-        named_args = tuple()
-        projections = tuple()
+        named_args = list()
+        projections = list()
```
Comment (tgy) [inline neurolang/datalog/translate_to_named_ra.py:178]: changed from `tuple` to `list` to make them mutable in recursive calls

File: neurolang/datalog/translate_to_named_ra.py [hunk 2]
```diff
@@ -174,25 +174,57 @@ def translate_builtin_fa_constants(self, expression):
...
-                projections += (Constant[ColumnInt](i, verify_type=False),)
-                named_args += (arg,)
-
+        TranslateToNamedRA._translate_fa_args(
+            expression.args,
+            named_args,
+            projections,
+            selections,
+            selection_columns,
+            [0],
```
Comment (tgy) [inline neurolang/datalog/translate_to_named_ra.py:187]: using a list here is my way of keeping a mutable counter in recursive calls (to make sure each named arg gets a different `ColumnInt` in selections)

Comment (demianw) [reply neurolang/datalog/translate_to_named_ra.py:187]: Why not using a stack-based algorithm and then just counter @tgy ?

Comment (tgy) [reply neurolang/datalog/translate_to_named_ra.py]: done, sorry i think my Caml kicked in again

Comment (demianw) [reply neurolang/datalog/translate_to_named_ra.py]: 🐫 🐫 🐪

**Comment by tgy** [comment]
Can confirm this solves the issue with the spherical volume region query-based probfact example

**Pull request merged**